### PR TITLE
syntax error fixed in fesom-2.1.yaml

### LINKED ISSUE #674: Problem with esm_master
**Describe the problem you are facing**
After installing the esm_tools on levante, I get an error when trying esm_master:
`Something else went wrong
Serious issue with /home/a/a270147/esm_tools/configs//components/fesom/fesom-2.1.yaml, goodbye...`

**System (please complete the following information):**
 - Supercomputer: Levante
 - Version: I'm not sure (there is no esm_versions anymore?) But I downloaded from 'origin/release' on May 30 15:07 if that helps.

### PR DESCRIPTION
**Pull request by mandresm**
Closes #674

### CODE REVIEW

**Comment by mandresm** [comment]
This should go in asap as the mistake is breaking `esm_master` in release.

**Comment by mandresm** [comment]
I have also added a clearer error so that instead of returning the confusing message:
```
Something else went wrong
Serious issue with /home/a/a270147/esm_tools/configs//components/fesom/fesom-2.1.yaml, goodbye...
```
it now returns a clearer message giving the pyyaml hints about the error:
```
ERROR: Yaml syntax                                                                           
------------------                                                                  
Syntax error in /home/a/a270152/esm_tools/configs//components/fesom/fesom-2.1.yaml               
                                                                                          
Details:                                                                                  
while parsing a block mapping                                                             
  in "/home/a/a270152/esm_tools/configs//components/fesom/fesom-2.1.yaml", line 63, column 9
expected <block end>, but found '<block mapping start>'                                       
  in "/home/a/a270152/esm_tools/configs//components/fesom/fesom-2.1.yaml", line 81, column 17

File: src/esm_parser/yaml_to_dict.py [hunk 1]
```diff
@@ -210,8 +210,9 @@ def yaml_file_to_dict(filepath):
             logger.debug(f"Your file {filepath + extension} has syntax issues!")
             raise EsmConfigFileError(filepath + extension, yaml_error)
         except Exception as error:
-            print("Something else went wrong")
-            print(f"Serious issue with {filepath}, goodbye...")
+            esm_parser.user_error(
+                "Yaml syntax",
+                f"Syntax error in ``{filepath}``\n\n``Details:\n``{error}")
             logger.exception(error)
```
Comment (denizural) [inline src/esm_parser/yaml_to_dict.py]: I think those 2 lines will never be reached since `user_error` already calls `sys.exit(exit_code)`

Comment (mandresm) [reply src/esm_parser/yaml_to_dict.py]: thanks, that's a very good point. I'll change that now.

**Review by denizural — APPROVED** [review]
There is a minor comment but I think it is not so important

**Comment by mandresm** [comment]
#bump

**Pull request merged**